### PR TITLE
Add get and set bucket CORS methods

### PR DESCRIPTION
**Pull request by markduan**

### CODE REVIEW

**Comment by osier** [comment]
The method to delete cors is missed, others looks good to me.

**Comment by osier** [comment]
Please add the method to delete cors. Thanks.

**Comment by markduan** [comment]
I just add the method to delete CORS, please accept this pull request.

File: qingcloud/qingstor/bucket.py [hunk 1]
```diff
@@ -191,6 +191,46 @@ def set_acl(self, acls):
...
+            return resp
+        else:
+            err = get_response_error(response)
+            raise err
+
+    def set_cors(self, cors):
+        """ Set the bucket CORS settings.
+
+        Keyword arguments:
+        cors - a dict object contains CORS rulses
```
Comment (prettyxw) [inline qingcloud/qingstor/bucket.py]: Found typo, "rulses" should be "rules".

File: qingcloud/qingstor/bucket.py [hunk 2]
```diff
@@ -191,6 +191,46 @@ def set_acl(self, acls):
...
+            return True
+        else:
+            err = get_response_error(response)
+            raise err
+
+    def delete_cors(self, cors):
+        """ Delete the bucket CORS settings.
+        """
+        params = {"cors": None}
+        response = self.connection.make_request("DELETE", self.name, params=params)
```
Comment (prettyxw) [inline qingcloud/qingstor/bucket.py]: PEP 8: line to long (83 > 80 characters)

File: qingcloud/qingstor/bucket.py [hunk 3]
```diff
@@ -191,6 +191,46 @@ def set_acl(self, acls):
...
+        if response.status == 200:
```
Comment (prettyxw) [inline qingcloud/qingstor/bucket.py]: HTTP return code should be 204 (No content) when delete successfully.

File: qingcloud/qingstor/bucket.py [hunk 4]
```diff
@@ -191,6 +191,46 @@ def set_acl(self, acls):
...
+        data = json.dumps(cors)
+        response = self.connection.make_request("PUT", self.name,
+                                                params=params, data=data)
+        if response.status == 200:
+            return True
+        else:
+            err = get_response_error(response)
+            raise err
+
+    def delete_cors(self, cors):
```
Comment (prettyxw) [inline qingcloud/qingstor/bucket.py]: Parameter "cors" is useless.

File: qingcloud/qingstor/bucket.py (hunk 2 above)
Comment (prettyxw) [inline qingcloud/qingstor/bucket.py]: ", params=params" can be removed.

**Comment by prettyxw** [comment]
Tests for those changes are missing, I will check that.

**Comment by prettyxw** [comment]
@markduan, for your reference

``` python
def test_bucket_get_cors(self):
    body = {
        "cors_rules": [
            {
                "allowed_origin": "http://*.qingcloud.com",
                "allowed_methods": [
                    "PUT",
                    "GET",
                    "DELETE",
                    "POST"
                ],
                "allowed_headers": [
                    "X-QS-Date",
                    "Content-Type",
                    "Content-MD5",
                    "Authorization"
                ],
                "max_age_seconds": 200,
                "expose_headers": [
                    "X-QS-Date"
                ]
            }
        ]
    }
    self.mock_http_response(status_code=200, body=json.dumps(body))
    acls = self.bucket.get_cors()
    self.assertDictEqual(acls, body)

def test_bucket_set_cors(self):
    self.mock_http_response(status_code=200)
    ret = self.bucket.set_cors({
        "cors_rules": [
            {
                "allowed_origin": "http://*.example.com",
                "allowed_methods": [
                    "PUT",
                    "GET",
                    "DELETE",
                    "POST"
                ],
                "allowed_headers": [
                    "*"
                ],
                "max_age_seconds": 400
            }
        ]
    })
    self.assertTrue(ret)

def test_bucket_delete_cors(self):
    self.mock_http_response(status_code=204)
    ret = self.bucket.delete_cors()
    self.assertTrue(ret)
```

**Comment by markduan** [comment]
@prettyxw check it again, please.

**Pull request merged**